android/services/mongodb-remote/src/androidTest/java/com/mongodb/stitch/android/services/mongodb/remote/internal/SyncMongoClientIntTests.kt

```diff
@@ -71,20 +76,36 @@ class SyncMongoClientIntTests : BaseStitchAndroidIntTest(), SyncIntTestRunner {
             sync.syncOne(id)
         }
 
-        override fun insertOneAndSync(document: Document): RemoteInsertOneResult {
+        override fun count(filter: Bson): Long {
+            return Tasks.await(sync.count())
```

Comment:
Even if not used currently, expected behavior would be for this to propagate the filter.
```suggestion
            return Tasks.await(sync.count(filter))
```